infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -124,7 +126,7 @@
     /**
      * Current node's id
      */
-    private Optional<String> localNodeId = Optional.empty();
+    private String localNodeId;
```

Comment:
instead of String, it should be URL. Would be great to change it to URL if possible

Reply:
The node id's format is more like a uuid, so I used String here.